src/styles/index.scss

```diff
@@ -1,3 +1,66 @@
 body {
   margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+$day-size: 100px;
+$gap: 1px;
+
+.calendar {
+  max-width: calc($day-size * 7 + $gap * 6);
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: 10px;
+  &__day {
+    background-color: #eee;
+    position: relative;
+    box-sizing: border-box;
+    width: $day-size;
+    height: $day-size;
+    border: 1px solid black;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    &:hover {
+      background-color: #ffbfcb;
+      transform: translateY(-20px);
+      cursor: pointer;
+      transition-duration: 0.5s;
+    }
+  }
+}
+
+@for $i from 1 to 32 {
+  .calendar :nth-child(#{$i}):after {
+    content: '#{$i}';
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+  }
+}
+
+$days: (
+  mon: 0,
+  tue: 1,
+  wed: 2,
+  thu: 3,
+  fri: 4,
+  sat: 5,
+  sun: 6,
+);
+
+@each $day, $index in $days {
+  .calendar--start-day-#{$day} :first-child {
+    margin-left: ($day-size + $gap) * $index;
+  }
```

Comment:
The selector `.calendar--start-day-#{$day} :first-child` may not work as intended due to the space before `:first-child`, which makes it a descendant combinator. This will select any first child of any element within `.calendar--start-day-#{$day}`. Consider removing the space to target the first child of `.calendar--start-day-#{$day}` directly.